be/src/exprs/aggregate_functions.cpp

```diff
@@ -231,6 +232,81 @@ void AggregateFunctions::avg_update(FunctionContext* ctx, const T& src, StringVa
     ++avg->count;
 }
 
+struct PercentileApproxState {
+public:
+    TDigest *digest;
+    double targetQuantile = -1.0;
```

Comment:
It's better to initialize digest to nullptr and delete digest in destructor function